### PR TITLE
Adding support for `component('button', {})` syntax for EJS

### PR DESCRIPTION
**Pull request by Robbie-Cook**
N.B. This PR should preserve the existing syntax too (as legacy) in the ejs-engine

Request from the app team.

 They had a few concerns about the existing syntax:

```ejs
      <%- include('components/input/input.ejs', { ...props });
```

Their concerns were:

* 'include', though it conforms to ejs's native syntax, doesn't do what a generic include would do

* The 'Component' prefix is not necessary, and too verbose

Here is our proposed syntax:

```ejs
<%- component('button/button.ejs', { text: props.text, level }); %>
```

### CODE REVIEW

**Comment by Robbie-Cook** [comment]
@bglw it won't let me make a PR to `patch-1`. Could we merge `patch-1` to master

**Comment by Robbie-Cook** [comment]
@bglw Good idea! Will do.

**Comment by bglw** [comment]
Here's the step in ruby to convert it. Path + last section of path + extension

https://github.com/CloudCannon/bookshop/blob/master/plugins/jekyll-bookshop/lib/jekyll-bookshop.rb#L37-L38

**Comment by Robbie-Cook** [comment]
@bglw  I think I have implemented this now!: https://github.com/Robbie-Cook/bookshop/commit/3997eaed90ef4adf181eaadf9dd8009d10783c39

Also, I am supporting the old import syntax just in case.

File: engines/ejs-engine/src/index.js [hunk 1]
```diff
@@ -31,10 +31,17 @@ const engine = {
  * @return {string} EJS component string
  */
 window.component = (path, props) => {
+  	// Handle that the full path is not provided (just component-name, e.g, component("button"))
+	const modifiedPathParts = path.split('/');
+	const lastPart = modifiedPathParts[modifiedPathParts.length - 1];
+  const newPath = `${path.toLowerCase().replace('.ejs', '')}/${lastPart}.ejs`;
```
Comment (bglw) [inline engines/ejs-engine/src/index.js:37]: Looks like lastPart will make it through with case preserved, you might want to lowercase that too

Comment (Robbie-Cook) [reply engines/ejs-engine/src/index.js:37]: Right, good spotting

Comment (Robbie-Cook) [reply engines/ejs-engine/src/index.js]: Fixed in https://github.com/CloudCannon/bookshop/pull/18/commits/a030c2a32dd3cd0e65792ebd6cf7fd57bc8939ee :)

**Review by bglw — APPROVED** [review]

**Comment by bglw** [comment]
Released as `1.6.2`

**Pull request merged**